students/will_chang/lesson09/mailroom_oo.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+
+import datetime
+import os.path
+import pathlib
+import sys
+
+class Donor:
+    """
+    A class used to represent a Donor
+    """
+    def __init__(self, name):
+        """
+        Parameters
+        ----------
+        name : str
+            The name of the Donor
+        donations : list
+            List of donations from the given Donor
+        """
+        self.name = name
+        self.donations = []
+    
+    def name(self):
+        return self.name
+        
+    def add_donation(self, donation):
+        """Add a donation to the list of donations"""
+        self.donations.append(donation)
+    
+    def total_donation(self):
+        """Return the total donation amount"""
+        return sum(self.donations)
+        
+    def num_donation(self):
+        """Return the quantity of donations"""
+        return len(self.donations)
+        
+    def avg_donation(self):
+        """Return the average donation amount"""
+        return sum(self.donations) / len(self.donations)       
+        
+        
+class DonorCollection:
+    """
+    A class used to represent a collection of Donors
+    """
+    def __init__(self):
+        """
+        Parameters
+        ----------
+        donor_list : list
+            List of donors with their associated total donation, number of donations, and average donation
+        """
+        self.donor_list = []
+        
+    def add_donor(self, donor):
+        """Add donor information to the donor list"""
+        self.donor_list.append([donor.name, donor.total_donation(), donor.num_donation(), donor.avg_donation()])
```

Comment:
your donor list should contain donor objects only, no need to extract all of the information out of them.

for example:
```
self.donor_list = [Donor("bob"), Donor("john")]
for donor in self.donor_list:
    print(donor.name)
```
makes a lot more sense that storing all the primitive data out of those objects.